### PR TITLE
liquid-ceph: add support for per-az usage and quota

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@clyso.com>

### CODE REVIEW

File: system/liquid-ceph/templates/configmap.yaml [hunk 1]
```diff
@@ -23,3 +23,10 @@ data:
 
     [client.admin]
     keyring = /etc/ceph-keyring/keyring
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: liquid-ceph-zone-mapping
+data:
+  zone-mapping.yaml: ""
```
Comment (Nuckal777) [inline system/liquid-ceph/templates/configmap.yaml:32]: I think the value here needs to be templated.

File: system/liquid-ceph/templates/deployment.yaml [hunk 2]
```diff
@@ -70,21 +71,13 @@ spec:
...
-              secretKeyRef:
-                name: {{ .Values.ceph.rgwSecretName }}
-                key: SecretKey
           securityContext:
             runAsNonRoot: true
           volumeMounts:
           - mountPath: /etc/liquid
             name: policy
+          - mountPath: /etc/liquid
+            name: zone-mapping
```
Comment (Nuckal777) [inline system/liquid-ceph/templates/deployment.yaml:80]: I think the mount points collide. I would propose to move `zone-mappings.yaml` and `policies.yaml` in a single ConfigMap (e.g. called just `liquid-ceph`). Then mounting that single ConfigMap would provide both files.

**Review by Nuckal777 — CHANGES_REQUESTED** [review]

File: system/liquid-ceph/templates/configmap.yaml (hunk 1 above)
Comment (majewsky) [inline system/liquid-ceph/templates/configmap.yaml:32]: Why do you try to make the configmap name the thing that is configured? The standard approach is to take the zoneMapping config itself via `values.yaml` and render it out here (Ctrl-F `toYaml` on this repo for many many examples of how to do this). Then hardcode the configMap name at the usage site.

**Review by majewsky — CHANGES_REQUESTED** [review]

File: system/liquid-ceph/templates/configmap.yaml (hunk 1 above)
Comment (databus23) [inline system/liquid-ceph/templates/configmap.yaml:32]: To directly specify the yaml in the values file and render it out here I think this should do it.
```suggestion
  zone-mapping.yaml: | {{- toYaml .Values.ceph.zoneMapping | nindent 4 }}
```

File: system/liquid-ceph/values.yaml [hunk 3]
```diff
@@ -3,13 +3,10 @@ image:
...
 ceph:
-  endpoint: "" # "http://ceph-rgw.ceph.svc.cluster.local:80"
-  targetZone: ""
-  validPools: "" # regex
-  resourceName: "objectstore-ceph-region-3-size"
   secretName: "" # ceph.conf and keyring file
-  rgwSecretName: "" # AccessKey and SecretKey fields
-  allowedDomains: [] 
+  allowedDomains: []
+  zoneMapping: "" # zone mapping file
```
Comment (databus23) [inline system/liquid-ceph/values.yaml:8]: file is misleading there needs to be yaml there
```suggestion
  zoneMapping: {} # zone mapping yaml
```

File: system/liquid-ceph/ci/test-values.yaml [hunk 4]
```diff
@@ -6,7 +6,8 @@ global:
   linkerd_requested: false
 ceph:
   secretName: ceph-secret
-  endpoint: "http://ceph-rgw.ceph.svc.cluster.local:80"
-  targetZone: the-zone
-  validPools: the-pools
-  rgwSecretName: rgw-secret
+  zoneMapping: zone-mapping
```
Comment (databus23) [inline system/liquid-ceph/ci/test-values.yaml:9]:
```suggestion
  zoneMapping:
    az-a-3:
      resource: local-standard
      az: a
```

File: system/liquid-ceph/templates/deployment.yaml [hunk 5]
```diff
@@ -39,17 +39,15 @@ spec:
           image: "{{required ".Values.global.registry is missing" .Values.global.registry }}/{{ .Values.image.repository }}:{{required ".Values.image.tag is missing" .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
+          {{ .Values.ceph.enableDebug }}
```
Comment (Nuckal777) [inline system/liquid-ceph/templates/deployment.yaml:42]: Missing `if`.

File: system/liquid-ceph/templates/deployment.yaml [hunk 6]
```diff
@@ -39,17 +39,15 @@ spec:
           image: "{{required ".Values.global.registry is missing" .Values.global.registry }}/{{ .Values.image.repository }}:{{required ".Values.image.tag is missing" .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
+          {{ if .Values.ceph.enableDebug }}
```
Comment (databus23) [inline system/liquid-ceph/templates/deployment.yaml:42]:
```suggestion
          {{- if .Values.ceph.enableDebug }}
```

File: system/liquid-ceph/templates/deployment.yaml [hunk 7]
```diff
@@ -39,17 +39,15 @@ spec:
...
+          - -d
+          {{ end }}
```
Comment (databus23) [inline system/liquid-ceph/templates/deployment.yaml:44]:
```suggestion
          {{- end }}
```

**Review by databus23 — APPROVED** [review]

File: system/liquid-ceph/ci/test-values.yaml [hunk 8]
```diff
@@ -6,7 +6,11 @@ global:
   linkerd_requested: false
 ceph:
   secretName: ceph-secret
-  endpoint: "http://ceph-rgw.ceph.svc.cluster.local:80"
-  targetZone: the-zone
-  validPools: the-pools
-  rgwSecretName: rgw-secret
+  zoneMapping:
+    az-a-3:
```
Comment (kayrus) [inline system/liquid-ceph/ci/test-values.yaml:10]: can we reuse the `rgwTargetPlacements` and `premiumPlacements` mappings from the system/cc-ceph chart? duplicating zone mappings adds flexibility, but also a complexity and a room for a human error.

Comment (kayrus) [reply system/liquid-ceph/ci/test-values.yaml:10]: cc @sumitarora2786

**Review by databus23 — APPROVED** [review]

**Pull request merged**